ddl/column.go

```diff
@@ -903,11 +927,24 @@ func updateChangingCol(col *model.ColumnInfo, newName model.CIStr, newOffset int
 	return col
 }
 
-func getOldIndexIDs(tblInfo *model.TableInfo, oldCol *model.ColumnInfo) []int64 {
+func getRelatedIndexInfos(tblInfo *model.TableInfo, colID int64) []*model.IndexInfo {
+	var indexInfos []*model.IndexInfo
+	for _, idx := range tblInfo.Indices {
+		for _, idxCol := range idx.Columns {
+			if tblInfo.Columns[idxCol.Offset].ID == colID {
+				indexInfos = append(indexInfos, idx)
+				break
+			}
+		}
+	}
+	return indexInfos
```

Comment:
Here need not use loop in loop.  Can we create the hashmap by ```tblInfo.Indices``` and ```idxCol``` can be checked in the hashmap.

Reply:
I think using loop is fine, because the length of `tableInfo.Indices` is less than `10` in 99% cases. For hashmap, the hash function calculation doesn't have the performance advantage.